src/main/java/appeng/menu/AEBaseMenu.java

```diff
@@ -825,4 +835,48 @@ public void handle(FriendlyByteBuf buffer) {
         }
     }
 
+    /**
+     * Can only be used with a host that implements {@link IEnergySource} only call once per broadcastChanges()
+     */
+    protected void updateItemPowerStatus() {
+        if ((guiItem instanceof IEnergySource energySource)) {
+            this.powerTicks++;
+            if (this.powerTicks > 10) {
+                energySource.extractAEPower(this.powerMultiplier * this.powerTicks, Actionable.MODULATE,
+                        PowerMultiplier.CONFIG);
+                this.powerTicks = 0;
+            }
+        }
+    }
+
+    /**
+     * Can only be used with a host that extends {@link WirelessTerminalGuiObject}
+     */
+    protected void checkWirelessRange() {
+        if (guiItem instanceof WirelessTerminalGuiObject wirelessTerminalGUIObject) {
+            if (!wirelessTerminalGUIObject.rangeCheck()) {
+                if (isServer() && this.isValidMenu()) {
+                    this.getPlayerInventory().player.sendMessage(PlayerMessages.OutOfRange.get(), Util.NIL_UUID);
+                }
+
+                this.setValidMenu(false);
+            } else {
+                powerMultiplier = AEConfig.instance().wireless_getDrainRate(wirelessTerminalGUIObject.getRange());
+            }
+        }
+    }
+
+    /**
+     * Can only be used with a host that implements {@link IInventorySlotAware}
+     */
+    protected boolean checkGuiItemNotInSlot() {
+        if (guiItem instanceof IInventorySlotAware iInventorySlotAware) {
+            if (!ensureGuiItemIsInSlot(guiItem, iInventorySlotAware.getInventorySlot())) {
+                this.setValidMenu(false);
+                return true;
+            }
+        }
+        return false;
+    }
+
```

Comment:
We'll need to find another home for this. AEBaseMenu is already a god-class, we should avoid putting this here